apps/backend/src/workflowEngine/index.ts

```diff
@@ -57,6 +59,67 @@ const shouldMoveToNextStatus = (
   return allNextStatusRulesFulfilled;
 };
 
+const checkIfConditionsForNextStatusAreMet = async ({
+  nextWorkflowConnections,
+  proposalWorkflow,
+  proposalSettingsDataSource,
+  proposalWithEvents,
+}: {
+  nextWorkflowConnections: ProposalWorkflowConnection[];
+  proposalWorkflow: ProposalWorkflow;
+  proposalSettingsDataSource: ProposalSettingsDataSource;
+  proposalWithEvents: {
+    proposalPk: number;
+    proposalEvents?: ProposalEventsRecord;
+    currentEvent: Event;
+  };
+}) => {
+  for (const nextWorkflowConnection of nextWorkflowConnections) {
+    if (!nextWorkflowConnection.nextProposalStatusId) {
+      continue;
+    }
+
+    const nextNextWorkflowConnections =
+      await getProposalWorkflowConnectionByStatusId(
+        proposalWorkflow.id,
+        nextWorkflowConnection.nextProposalStatusId
+      );
+    const newStatusChangingEvents =
+      await proposalSettingsDataSource.getStatusChangingEventsByConnectionIds(
+        nextNextWorkflowConnections.map((connection) => connection.id)
+      );
+
+    if (!proposalWithEvents.proposalEvents) {
+      return;
+    }
+
+    newStatusChangingEvents.forEach((sce) => {
+      const proposalEventsKeys = Object.keys(
+        proposalWithEvents.proposalEvents!
+      );
+      const allProposalCompleteEvents = proposalEventsKeys.filter(
+        (proposalEventsKey) =>
+          proposalWithEvents.proposalEvents![
+            proposalEventsKey as keyof ProposalEventsRecord
+          ]
+      );
+
+      const nextStatusRulesFulfilled = allProposalCompleteEvents.includes(
+        sce.statusChangingEvent.toLowerCase()
+      );
+
+      if (sce.statusChangingEvent && nextStatusRulesFulfilled)
+        workflowEngine([
```

Comment:
```suggestion
        await workflowEngine([
```

Reply:
@janosbabik I think we don't really want to await this and block the running of the other code because of it. The idea was that the whole workflow runs in the background and not blocking the main thread.